### PR TITLE
Evaluate the 128G RCAs only if CMS is the collector for OldGen

### PR DESCRIPTION
**Pull request by ktkrg**
*Fixes #:*

*Description of changes:*
This change adds a new metric node(GC_Type) which tells what collector is used for a memory pool. The metric is used as a gating mechanism to proceed with the evaluation of the HighOldGenOccupancyRca and OldGenReclamationRca.

*Tests:*
UTs, tested on docker.

*If new tests are added, how long do the new ones take to complete*

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: src/main/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/rca/store/ElasticSearchAnalysisGraph.java [hunk 1]
```diff
@@ -125,22 +126,27 @@ public void construct() {
...
             MetricsDB.AVG, CommonDimension.OPERATION.toString());
 
     heapUsed.addTag(TAG_LOCUS, LOCUS_DATA_MASTER_NODE);
     gcEvent.addTag(TAG_LOCUS, LOCUS_DATA_MASTER_NODE);
+    gcType.addTag(TAG_LOCUS, LOCUS_DATA_MASTER_NODE);
     heapMax.addTag(TAG_LOCUS, LOCUS_DATA_MASTER_NODE);
     gc_Collection_Time.addTag(TAG_LOCUS, LOCUS_DATA_MASTER_NODE);
     cpuUtilizationGroupByOperation.addTag(TAG_LOCUS, LOCUS_DATA_MASTER_NODE);
 
+
```
Comment (sidheart) [inline src/main/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/rca/store/ElasticSearchAnalysisGraph.java:141]: nit: unnecessary

File: src/main/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/rca/store/ElasticSearchAnalysisGraph.java [hunk 2]
```diff
@@ -125,22 +126,27 @@ public void construct() {
...
     addLeaf(heapUsed);
     addLeaf(gcEvent);
+    addLeaf(gcType);
     addLeaf(heapMax);
     addLeaf(gc_Collection_Time);
     addLeaf(cpuUtilizationGroupByOperation);
 
+
```
Comment (sidheart) [inline src/main/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/rca/store/ElasticSearchAnalysisGraph.java:149]: nit: unnecessary

File: src/main/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/rca/store/ElasticSearchAnalysisGraph.java [hunk 3]
```diff
@@ -176,12 +182,13 @@ public void construct() {
     hotNodeClusterRca.addTag(TAG_LOCUS, LOCUS_MASTER_NODE);
     hotNodeClusterRca.addAllUpstreams(Collections.singletonList(hotJVMNodeRca));
 
-    final HighOldGenOccupancyRca oldGenOccupancyRca = new HighOldGenOccupancyRca(heapMax, heapUsed);
+    final HighOldGenOccupancyRca oldGenOccupancyRca = new HighOldGenOccupancyRca(heapMax,
+        heapUsed, gcType);
     oldGenOccupancyRca.addTag(TAG_LOCUS, LOCUS_DATA_MASTER_NODE);
     oldGenOccupancyRca.addAllUpstreams(Arrays.asList(heapMax, heapUsed));
```
Comment (sidheart) [inline src/main/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/rca/store/ElasticSearchAnalysisGraph.java:188]: Add gcType as an upstream

File: src/main/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/rca/store/ElasticSearchAnalysisGraph.java [hunk 4]
```diff
@@ -176,12 +182,13 @@ public void construct() {
...
 
     final OldGenReclamationRca oldGenReclamationRca = new OldGenReclamationRca(heapUsed,
-        heapMax, gcEvent);
+        heapMax, gcEvent, gcType);
     oldGenReclamationRca.addTag(TAG_LOCUS, LOCUS_DATA_MASTER_NODE);
     oldGenReclamationRca.addAllUpstreams(Arrays.asList(heapUsed, heapMax, gcEvent));
```
Comment (sidheart) [inline src/main/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/rca/store/ElasticSearchAnalysisGraph.java:193]: Add gcType as an upstream

**Review by sidheart — CHANGES_REQUESTED** [review]

File: src/main/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/rca/store/ElasticSearchAnalysisGraph.java (hunk 3 above)
Comment (ktkrg) [reply src/main/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/rca/store/ElasticSearchAnalysisGraph.java]: good catch! Added.

File: src/main/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/rca/store/ElasticSearchAnalysisGraph.java (hunk 1 above)
Comment (ktkrg) [reply src/main/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/rca/store/ElasticSearchAnalysisGraph.java]: removed

**Review by sidheart — DISMISSED** [review]

**Review by yojs — APPROVED** [review]

**Review by sidheart — DISMISSED** [review]

**Review by yojs — DISMISSED** [review]

**Review by yojs — APPROVED** [review]

**Review by sidheart — APPROVED** [review]

**Pull request merged**